server/core/src/client_api_allowlist.cpp

```diff
@@ -8,193 +8,191 @@
+            SUB_STRUCT_FILE_RMDIR_AN,
+            SUB_STRUCT_FILE_OPENDIR_AN,
+            SUB_STRUCT_FILE_READDIR_AN,
+            SUB_STRUCT_FILE_CLOSEDIR_AN,
+            DATA_OBJ_TRUNCATE_AN,
+            SUB_STRUCT_FILE_TRUNCATE_AN,
+            //GET_XMSG_TICKET_AN,
+            //SEND_XMSG_AN,
+            //RCV_XMSG_AN,
+            SUB_STRUCT_FILE_GET_AN,
+            SUB_STRUCT_FILE_PUT_AN,
+            SYNC_MOUNTED_COLL_AN,
+            STRUCT_FILE_SYNC_AN,
+            CLOSE_COLLECTION_AN,
+            STRUCT_FILE_EXTRACT_AN,
+            STRUCT_FILE_EXT_AND_REG_AN,
+            STRUCT_FILE_BUNDLE_AN,
+            CHK_OBJ_PERM_AND_STAT_AN,
+            GET_REMOTE_ZONE_RESC_AN,
+            DATA_OBJ_OPEN_AND_STAT_AN,
+            //L3_FILE_GET_SINGLE_BUF_AN,
+            L3_FILE_PUT_SINGLE_BUF_AN,
+            DATA_OBJ_CREATE_AND_STAT_AN,
+            DATA_OBJ_CLOSE_AN,
+            DATA_OBJ_LSEEK_AN,
+            DATA_OBJ_READ_AN,
+            DATA_OBJ_WRITE_AN,
+            COLL_REPL_AN,
+            OPEN_COLLECTION_AN,
+            RM_COLL_AN,
+            MOD_COLL_AN,
+            COLL_CREATE_AN,
+            DATA_OBJ_UNLOCK_AN,
+            REG_COLL_AN,
+            PHY_BUNDLE_COLL_AN,
+            UNBUN_AND_REG_PHY_BUNFILE_AN,
+            GET_HOST_FOR_PUT_AN,
+            GET_RESC_QUOTA_AN,
+            BULK_DATA_OBJ_REG_AN,
+            BULK_DATA_OBJ_PUT_AN,
+            PROC_STAT_AN,
+            //STREAM_READ_AN,
+            //EXEC_CMD_AN,
+            //STREAM_CLOSE_AN,
+            GET_HOST_FOR_GET_AN,
+            DATA_OBJ_REPL_AN,
+            DATA_OBJ_COPY_AN,
+            DATA_OBJ_PHYMV_AN,
+            DATA_OBJ_FSYNC_AN,
+            DATA_OBJ_LOCK_AN,
+
+            GET_MISC_SVR_INFO_AN,
+            GENERAL_ADMIN_AN,
+            GEN_QUERY_AN,
+            AUTH_REQUEST_AN,
+            AUTH_RESPONSE_AN,
+            AUTH_CHECK_AN,
+            MOD_AVU_METADATA_AN,
+            MOD_ACCESS_CONTROL_AN,
+            RULE_EXEC_MOD_AN,
+            GET_TEMP_PASSWORD_AN,
+            GENERAL_UPDATE_AN,
+            READ_COLLECTION_AN,
+            USER_ADMIN_AN,
+            //GENERAL_ROW_INSERT_AN,
+            //GENERAL_ROW_PURGE_AN,
+            //END_TRANSACTION_AN,
+            //DATABASE_RESC_OPEN_AN,
+            //DATABASE_OBJ_CONTROL_AN,
+            //DATABASE_RESC_CLOSE_AN,
+            SPECIFIC_QUERY_AN,
+            TICKET_ADMIN_AN,
+            GET_TEMP_PASSWORD_FOR_OTHER_AN,
+            PAM_AUTH_REQUEST_AN,
+            GET_LIMITED_PASSWORD_AN,
+
+            SSL_START_AN,
+            SSL_END_AN,
+
+            AUTH_PLUG_REQ_AN,
+            AUTH_PLUG_RESP_AN,
+            GET_HIER_FOR_RESC_AN,
+            GET_HIER_FROM_LEAF_ID_AN,
+            SET_RR_CTX_AN,
+            EXEC_RULE_EXPRESSION_AN,
+
+            SERVER_REPORT_AN,
+            ZONE_REPORT_AN,
+            CLIENT_HINTS_AN
+        };
+        // clang-format on
+    } // init
+
+    auto enforce(const RsComm& _comm) noexcept -> bool
```

Comment:
This function is marked as `noexcept`, but can throw.

Either wrap the body in a try-catch block or remove the `noexcept` qualifier.